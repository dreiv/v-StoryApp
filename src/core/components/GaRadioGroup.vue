<script setup lang="ts">
import { computed, provide, useCssModule } from 'vue'
import { uniqueId } from '../utils/uniqueId'
import { radioGroupKey } from '../constants'

export interface RadioGroupProps {
  title?: string
  name?: string
  direction?: 'horizontal' | 'vertical'
}

const { direction = 'vertical', name = uniqueId('group') } = defineProps<RadioGroupProps>()
const style = useCssModule()

const classes = computed(() => [style.content, { [style.horizontal]: direction === 'horizontal' }])

provide(radioGroupKey, { name })
</script>

<template>
  <fieldset :class="$style.group">
    <legend :class="$style.title">
      <slot name="title">{{ title }}</slot>
    </legend>

    <div :class="classes">
      <slot />
    </div>
  </fieldset>
</template>

<style module>
.group {
  margin: 0;
  border: none;
  padding: 0;

  .title {
    margin-bottom: var(--ga-size-8);
    padding-left: 0;

    font-weight: var(--font-weight-medium);
    line-height: var(--text-md--line-height);
    font-family: var(--default-font-family);
  }

  .content {
    display: flex;
    flex-direction: column;
    gap: var(--ga-size-8);

    &.horizontal {
      flex-direction: row;
    }
  }
}
</style>
