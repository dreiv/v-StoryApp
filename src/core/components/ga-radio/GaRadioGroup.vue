<script setup lang="ts">
import { computed, provide, useCssModule } from 'vue'
import { radioGroupKey } from '@/core/constants'
import { uniqueId } from '@/core/utils/uniqueId'

export interface RadioGroupProps {
  title?: string
  name?: string
  direction?: 'horizontal' | 'vertical'
}

const { direction = 'vertical', name = uniqueId('group') } = defineProps<RadioGroupProps>()
const model = defineModel<string>({ default: '' })
const style = useCssModule()

const classes = computed(() => [style.content, { [style.horizontal]: direction === 'horizontal' }])

provide(radioGroupKey, { name, model })
</script>

<template>
  <fieldset :class="$style.group">
    <legend :class="$style.title">
      <slot name="title">{{ title }}</slot>
    </legend>

    <div :class="classes">
      <slot />
    </div>
  </fieldset>
</template>

<style module>
.group {
  margin: 0;
  border: none;
  padding: 0;
}

.title {
  padding-left: 0;

  font-weight: var(--font-weight-medium);
  line-height: var(--text-md--line-height);
  font-family: var(--default-font-family);

  &:not(:empty) {
    margin-bottom: var(--ga-size-spacing-03);
  }
}

.content {
  display: flex;
  flex-direction: column;
  gap: var(--ga-size-spacing-03);

  &.horizontal {
    flex-direction: row;
  }
}
</style>
