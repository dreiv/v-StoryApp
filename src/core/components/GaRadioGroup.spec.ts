import { defineComponent, inject } from 'vue'
import { mount, VueWrapper } from '@vue/test-utils'
import GaRadioGroup from './GaRadioGroup.vue'

vi.mock('../constants', () => ({ radioGroupKey: 'radioGroup' }))

const MockRadioConsumer = defineComponent({
  template: '<div>{{ injectedName }} - {{ injectedModel }}</div>',
  setup() {
    const group = inject<{ name: string; model: { value: string } }>('radioGroup')

    return {
      injectedName: group?.name,
      injectedModel: group?.model.value,
    }
  },
})

describe('GaRadioGroup', () => {
  let wrapper: VueWrapper
  beforeAll(() => {
    wrapper = mount(GaRadioGroup, { slots: { default: MockRadioConsumer } })
  })

  it('should render correctly', async () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render with horizontal direction', async () => {
    await wrapper.setProps({ direction: 'horizontal' })

    expect(wrapper.find('.horizontal')).toBeTruthy()
  })
})
