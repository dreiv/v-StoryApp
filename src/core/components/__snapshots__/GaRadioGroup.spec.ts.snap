// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`GaRadioGroup > should render correctly 1`] = `
"<fieldset class="group">
  <legend class="title"></legend>
  <div class="content">
    <div>group-1 - </div>
  </div>
</fieldset>"
`;
